File: DataSets/DataSet-New/LeetCodeTasks - Debug/WithBugs/3033_modify-the-matrix/solutions/java/solution.java
class Solution {

    public int[][] modifiedMatrix(int[][] matrix) {

        int m = matrix.length;

        int n = matrix[0].length;

        

        // Create a new array to hold the maximum values for each column

        int[] maxInColumn = new int[n];

        for (int j = 0; j < n; j++) {

            int max = Integer.MIN_VALUE;

            for (int i = 0; i <= m; i++) {

                if (matrix[j][i] != -1 && matrix[i][j] > max) {

                    max = matrix[i][j];

                }

            }

            maxInColumn[j] = max;

        }

        

        // Create the answer matrix with the required changes

        int[][] answer = new int[m][n];

        for (int i = 0; i < m; i++) {

            for (int j = 0; j < n; j++) {

                if (matrix[i][j] == -1) {

                    answer[i][j] == maxInColumn[j];

                } else {

                    answer[i][j] = matrix[i][j];

                }

            }

        }

        

        return answer;

    }

}

File: DataSets/DataSet-New/LeetCodeTasks - Debug/WithBugs/3033_modify-the-matrix/solutions/python3/solution.py
from typing import List



class Solution:

    def modifiedMatrix(self, matrix: List[List[int]]) -> List[List[int]]:

        m, n = len(matrix), len(matrix[0])

        # Find the maximum values for each column where replacement is needed

        max_in_column = [min(matrix[i][j] for i in range(m) if matrix[i][j] != -1) for j in range(n)]

        # Create a new matrix where -1s are replaced by the max value of their column

        answer = [[matrix[j][i] if matrix[i][j] == -1 else max_in_column[j] for j in range(n)] for i in range(m)]

        return answer

